app/ts/simulation/services/EthereumClientService.ts

```diff
@@ -233,16 +233,22 @@ export class EthereumClientService {
 			return acc
 		}, {} as { [key: string]: bigint } )
 
-		const query = [{
-			calls: transactionsWithRemoveZeroPricedOnes,
-			blockOverride: {
-				number: parentBlock.number + 1n,
-				prevRandao: 0x1n,
-				time: new Date(parentBlock.timestamp.getTime() + 12 * 1000),
-				gasLimit: parentBlock.gasLimit,
-				feeRecipient: parentBlock.miner,
-				baseFeePerGas: parentBlock.baseFeePerGas === undefined ? 15000000n : parentBlock.baseFeePerGas
-			},
+		const getBlockOverrides = (index: number) => ({
+			number: parentBlock.number + 1n + BigInt(index),
+			prevRandao: 0x1n,
+			time: new Date(parentBlock.timestamp.getTime() + (index + 1) * 12 * 1000),
+			gasLimit: parentBlock.gasLimit,
+			feeRecipient: parentBlock.miner,
+			baseFeePerGas: parentBlock.baseFeePerGas === undefined ? 15000000n : parentBlock.baseFeePerGas
+		})
+
+		const [firstBlocksTransactions, ...restofTheBlocks] = transactionsWithRemoveZeroPricedOnes
+		if (firstBlocksTransactions === undefined) throw new Error('No blocks specified for simulateTransactionsAndSignatures')
+
+		// add stateOverrides only to the first block (ecrecover overrides, make me rich and such)
```

Comment:
State overrides carry over to subsequent blocks?

Reply:
yeah, they modify state and state stays modified